src-tauri/src/overlay_server.rs

```diff
@@ -1,37 +1,34 @@
-use std::fs::File;
-use std::panic;
 use log::{error, info};
+use std::fs::File;
 use std::path::PathBuf;
 
-use tiny_http::{Server, Response, StatusCode};
-
-pub fn run_http_server(streamer_overlay_path: PathBuf) {
-
-    let result = panic::catch_unwind(|| {
+use tiny_http::{Response, Server, StatusCode};
 
-       // Ideally we would allow setting up port in the settings
-       info!("Starting streamer overlay server on port 47824");
-       let server = Server::http("127.0.0.1:47824").unwrap();
+use crate::config::OVERLAY_PORT;
 
-       for request in server.incoming_requests() {
-
-           let file = match File::open(&streamer_overlay_path) {
-               Ok(file) => file,
-               Err(_) => {
-                   let response = Response::new_empty(StatusCode(404));
-                   let _ = request.respond(response);
-                   continue;
-               }
-           };
-
-           let response = Response::from_file(file);
-           let _ = request.respond(response);
-       }
-
-    });
+pub fn run_http_server(streamer_overlay_path: PathBuf) {
+    // Ideally we would allow setting up port in the settings
+    info!("Starting streamer overlay server on port {}", OVERLAY_PORT);
 
-     if let Err(err) = result {
+    let server = match Server::http(format!("127.0.0.1:{}", OVERLAY_PORT)) {
+        Ok(server) => server,
+        Err(err) => {
             error!("Couldn't start the streamer overlay server: {:?}", err);
-      }
-
+            return;
```

Comment:
Not necessary to do here I'd say, but it probably makes sense to find all the `error!` log calls and send them to your monitoring tool too eventually.